src/dotnet/Chat.Service/Chats.cs

```diff
@@ -352,6 +354,33 @@ public virtual async Task RemoveTextEntry(IChats.RemoveTextEntryCommand command,
             await RemoveChatEntry(session, chatId, textEntry.AudioEntryId.Value, ChatEntryType.Audio, cancellationToken).ConfigureAwait(false);
     }
 
+    public virtual async Task LeaveChat(IChats.LeaveChatCommand command, CancellationToken cancellationToken)
+    {
+        if (Computed.IsInvalidating())
+            return; // It just spawns other commands, so nothing to do here
+        var (session, chatId) = command;
+        // NOTE(DF): may be replace with another membership check
+        var chatAuthor = await ChatAuthors.GetOwnAuthor(session, chatId, cancellationToken).ConfigureAwait(false);
+        if (chatAuthor == null || chatAuthor.HasLeft)
+            return;
+        var chat = await Get(session, chatId, cancellationToken).ConfigureAwait(false);
+        if (chat == null)
+            return;
+
+        var user = await Auth.GetUser(session, cancellationToken).ConfigureAwait(false);
+        if (chat.OwnerIds.Contains(user.Id)) {
+            throw new InvalidOperationException(
+                    "You are the last owner of the chat. At the moment delete and leave feature is missing.");
```

Comment:
InvalidOperationException -> NotSupportedException, + text = "The very last owner of the chat can't leave it."

Reply:
fixed